pkg/controller/plugins.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/golang/glog"
+)
+
+// ControllerHost is an interface that plugins can use to access the kube-controller-manager
+type ControllerHost interface {
+	// GetKubeClient returns a client interface
+	GetKubeClient() client.Interface
+
+	// SyncPeriod asks the host for the sync period of the controller plugin with the given name.
+	// An error is returned if a period does not exist for the given name.
+	SyncPeriod(name string) (*time.Duration, error)
+
+	// SetSyncPeriod allows the plugin to set a default period with the host, also allows the host to override.
```

Comment:
How does the host override?  By just not accepting this new value and returning its own value in SyncPeriod?  Maybe this should return bool or error if the host does not accept the value?  Or can the host vary the value dynamically (and therefore have no hook to tell the plugin) ?  Once you decide, comments!